plonk/src/proof_system/prover.rs

```diff
@@ -919,6 +921,29 @@ impl<E: PairingEngine> Prover<E> {
                 )
             })
             .collect();
+
+        // mask splitting polynomials
+        // t'(X) = t(X) - b_last + b_now * X^(n+2)
+        // with t'_lowest(X) = t_lowest(X) - 0 + b_now * X^(n+2)
+        // and t'_highest(X) = t_highest(X) - b_last
+        let mut last_randomizer = E::Fr::zero();
+        split_quot_polys
+            .iter_mut()
+            .take(num_wire_types - 1)
+            .for_each(|poly| {
+                let now_randomizer = E::Fr::rand(prng);
```

Comment:
This is masking blinding factors that supposed to be only known to prover, not public coins from the transcript, otherwise we lost zk. 

So yeah it should be randomly sampled. (You can also look at how we mask wire polynomial as a reference)